kotest-core/src/commonMain/kotlin/io/kotest/core/test/TestName.kt

```diff
@@ -31,18 +32,40 @@ data class TestName(val prefix: String?, val name: String, val focus: Boolean, v
     *
     * This means the user writes when("!disable") and the platform invokes ("when", "!disable")
     * and ends up with !when disable, so that it is correctly parsed by the test runtime.
+    *
+    * To create test names with proper case, this function's result is affected by the
+    * [Project.testNameCase] property.
     */
    fun displayName(): String {
       val flattened = name.trim().replace("\n", "")
       val withPrefix = if (Project.includeTestScopePrefixes() && prefix != null) prefix else ""
+
+      val name = if (withPrefix.isBlank()) {
+         when (Project.testNameCase()) {
+            TestNameCaseOptions.SENTENCE -> flattened.capitalize()
+            TestNameCaseOptions.LOWERCASE -> flattened.uncapitalize()
+            else -> flattened
+         }
+      }
+      else {
+         when (Project.testNameCase()) {
+            TestNameCaseOptions.SENTENCE -> "${withPrefix.capitalize()}${flattened.uncapitalize()}"
+            TestNameCaseOptions.LOWERCASE -> "${withPrefix.uncapitalize()}${flattened.uncapitalize()}"
```

Comment:
This would leave mySILLYTESTNAME as mySILLYTESTNAME which doesn't seem intuitive with lowercase.

Reply:
Right! Would START_LOWERCASE be clearer? Any suggestion?

Reply:
Yes, or INITIAL_LOWER or something.
And I would keep LOWERCASE as a fourth option that just does .tolowercase on it all.

Reply:
One final thing, can you call them Lowercase, Sentence etc as that fits in with the enum style used everywhere else.

Eg,

enum class TestCaseOrder {
   // the order in which the tests are defined in code
   Sequential,
   // a randomized order
   Random,
   // ordered a-z
   Lexicographic
}

Reply:
Ok, I've applied the suggested changes